packages/table/src/cell/editableCell.tsx

```diff
@@ -49,30 +49,42 @@ export interface IEditableCellProps extends ICellProps {
 }
 
 export interface IEditableCellState {
-    isEditing: boolean;
+    isEditing?: boolean;
+    savedValue?: string;
+    dirtyValue?: string;
 }
 
 export class EditableCell extends React.Component<IEditableCellProps, IEditableCellState> {
-    public state = {
-        isEditing: false,
-    };
+    public state: IEditableCellState;
+
+    public constructor(props: IEditableCellProps, context?: any) {
+        super(props, context);
+        this.state = {
+            isEditing: false,
+            savedValue: props.value,
+        };
+    }
 
     public shouldComponentUpdate(nextProps: IEditableCellProps, nextState: IEditableCellState) {
         return !Utils.shallowCompareKeys(this.props, nextProps, { exclude: ["style"] })
             || !Utils.shallowCompareKeys(this.state, nextState)
             || !Utils.deepCompareKeys(this.props, nextProps, ["style"]);
     }
 
+    public componentWillReceiveProps(nextProps: IEditableCellProps) {
+        const { value } = nextProps;
+        this.setState({ savedValue: value, dirtyValue: value });
```

Comment:
If I'm actively editing the value and the props change, is it best to replace the "dirty" value I'm editing, pulling the rug out from under me, or is it best to update only the saved value, giving no visual indication to the user yet that the saved value as changed?